src/Keywords/KeywordsInterface.php

```diff
@@ -18,13 +18,17 @@
  * @phpstan-type TLanguage string
  * @phpstan-type TKeywordsArray array{feature: string, background: string, scenario: string, scenario_outline: string, examples: string, given: string, when: string, then: string, and: string, but: string}
  * @phpstan-type TMultiLanguageKeywords array<TLanguage, TKeywordsArray>
+ * @phpstan-type TKeywordsString string
+ * @phpstan-type TStepKeywordsType 'Given'|'When'|'Then'|'And'|'But'
```

Comment:
I'm wondering whether we should use such a type alias here or define a `StepKeywordType` class with constants for each type and then use `StepKeywordType::*` as type in place needing it (while also being able to use the constants instead of literal strings in places

Reply:
I was actually wondering whether we should introduce an enum for this now. 

We may be able to use the actual enum internally in some places, and where we still need to use strings the phpstan `value-of<StepKeywordType>` would work the same as `StepKeywordType::*`. I think there are also some methods with parameters that could be `StepKeywordType|value-of<StepKeywordType>` where we could safely accept either and convert internally as required. 

That would get us phpstan and runtime benefits from using an enum now without a BC break, and provide a cleaner path for making more use of the enum in a future major?

Reply:
True that we can't return the enum at runtime in 4.x, however if we are making that class final we could easily do so in 5.x. - and we then only change internal API rather than also changing a (public) const into an enum?